File: 01_Backend/EduMS.Backend/src/EduMS.Domain/Entities/M1_SchoolAdmin/ClassroomOperationalRule.cs
using EduMS.Domain.Common;

namespace EduMS.Domain.Entities;

public class ClassroomOperationalRule : BaseAuditableEntity
{
    public long ClassroomId { get; set; }
    public string RuleCode { get; set; } = string.Empty;
    public string RuleTitleAr { get; set; } = string.Empty;
    public string RuleTitleEn { get; set; } = string.Empty;
    public decimal MaxAllowedAbsencePercentage { get; set; } = 15.0m;
    public bool RequiresDailyAttendanceLog { get; set; } = true;
    public int AllowLateArrivalMinutes { get; set; } = 15;
    public int MaxAllowedConsecutiveAbsenceDays { get; set; } = 5;
    public int PenaltyTypeForExceedingLimit { get; set; } = 1; // 1=Warning, 2=Suspension, 3=ParentCall
    public DateTime? EffectiveStartDate { get; set; }
    public bool IsActive { get; set; } = true;

    // Navigation Property
    public virtual Classroom? Classroom { get; set; }
}

File: 04_Frontend_Application/Edums-Frontend/src/app/modules/m1-school-office/interface/classroom-operational-rule.ts
import type { RecordStatus, PenaltyType } from './common';

export interface ClassroomOperationalRule {
  id: number;
  classroomId: number;
  ruleCode: string;
  ruleTitleAr: string;
  ruleTitleEn: string;
  maxAllowedAbsencePercentage: number;
  requiresDailyAttendanceLog: boolean;
  allowLateArrivalMinutes: number;
  maxAllowedConsecutiveAbsenceDays: number;
  penaltyTypeForExceedingLimit: PenaltyType;
  effectiveStartDate: string | null;
  isActive: boolean;
}

export type CreateClassroomOperationalRuleDto = Omit<ClassroomOperationalRule, 'id' | 'isActive'>;

export type UpdateClassroomOperationalRuleDto = Omit<ClassroomOperationalRule, 'isActive'>;
